### PR TITLE
Fixes #1213 Size of image reduces on rotation several times

### LINKED ISSUE #1213: Size of image reduces on rotation several times
**Actual Behaviour**

On rotating the image and saving it multiple times the size of image keeps on reducing.

**Expected Behaviour**

The size of image should not be affected on rotating image.

**Steps to reproduce it**

Open the app, start editing and move to Transform edit mode. Rotate the image and apply the changes. Repeat this several times and the size of image will keep on reducing.

**Screenshots of the issue**
1.
screenshot_2017-09-25-20-50-38-126_org fossasia phimpme

2.
screenshot_2017-09-25-21-02-17-765_org fossasia phimpme

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by nb2998**
Fix #1213 

Changes: Size of images does not reduce now, instead gets cropped. To avoid cropping in case user wants simple rotation, a seperate button for 90 degree rotation has been added.

Screenshots for the change: 

screenshot_2017-09-28-19-25-20-645_org fossasia phimpme

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/RotateFragment.java [hunk 1]
```diff
@@ -77,8 +86,12 @@ public void onDetach() {
         resetRotateView();
     }
 
+    private void RotateRight() {
```
[the remark this answers is not in the document]
Comment (anantprsd5) [reply app/src/main/java/org/fossasia/phimpme/editor/fragment/RotateFragment.java]: @nb2998 Fix the codacy issue.

File: app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java [hunk 2]
```diff
@@ -75,29 +75,31 @@ public void draw(Canvas canvas) {
...
 			return;
-		maxRect.set(0, 0, getWidth(), getHeight());// The maximum bounding rectangle
-
+		maxRect.set(0, 0, (int)originImageRect.width(), (int)originImageRect.height());// The maximum bounding rectangle
 		calculateWrapBox();
 		scale = 1;
 		if (wrapRect.width() > getWidth()) {
-			scale = getWidth() / wrapRect.width();
-		}
+            scale=getWidth()/wrapRect.width();
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java]: Indent the codes properly.

Comment (nb2998) [reply app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java]: @anantprsd5 Yes, I'll do that. I wanted to know if the implementation for rotation is fine now or should be changed further?

**Comment by nb2998** [comment]
@anantprsd5 I feel cropping to be the only solution. Even google photos crops the picture to some extent on rotation. 
But instead of a seekbar, they have something like Horizontal Scroll View which I have tried to do in PR #1302

**Comment by anantprsd5** [comment]
Cropping the photos like this multiple times takes polygon kind of shape which does not happen in google photos as they are cropping it in dif. manner. If its possible, please try to implement something like that or else I like the previous implementation of decreasing the resolution as compared to this as of now. Thanks.

**Comment by nb2998** [comment]
@mohitmanuja As the implementation has changed a lot, will it be better to close the PR?

**Comment by anantprsd5** [comment]
@nb2998 Yes. Please close this PR.

**Pull request closed without merging**